src/core/lib/layer.js

```diff
@@ -589,7 +593,17 @@ ${flags.viewportChanged ? 'viewport' : ''}\
   // can be update correctly with minimal effort
   // TODO - arguments for testing only
   diffProps(newProps = this.props, oldProps = this.oldProps) {
-    const changeFlags = diffProps(newProps, oldProps, this._onUpdateTriggered.bind(this));
+    const changeFlags = diffProps(newProps, oldProps);
+
+    // iterate over changedTriggers
+    if (changeFlags.updateTriggersChanged) {
+      for (const key in changeFlags.updateTriggersChanged) {
+        if (changeFlags.updateTriggersChanged[key]) {
+          this._onUpdateTriggered(key);
```

Comment:
The `onUpdateTriggered` name was appropriate when this was a key. 

Now I think the code in that function should either be copied in here, or the helper function should be renamed to something less "callbacky", maybe `_activateUpdateTrigger()`